src/hooks/useQuizCategories.ts

```diff
@@ -0,0 +1,15 @@
+import { CategoriesType } from "../types"
+
+export type CategoryResponse = CategoriesType[];
+
+export const useQuizCategories = async (): Promise<CategoryResponse> => {
+    const response = await fetch(`https://opentdb.com/api_category.php`);
+    if (!response.ok) {
+      throw new Error("Problem fetching data");
+    }
+    const data = await response.json();
+  
+    return data.trivia_categories.map((category: CategoriesType) => ({
+        ...category
+    }));
+  };
```

Comment:
⚠️ I see where you're going with this, but if we put "effectful" code inside what looks like a custom hook without using `useEffect` to control that effect it will cause this request to be made each time the component that calls it re-renders—potentially hundreds of times in rapid succession. 😵

In its current state, we aren't using this function as a hook really, but more like a request function and passing it to `useQuery` so this is probably working in its current state without the issues I mentioned above. That said, if it is just abstracting the request functionality then we should likely avoid using the `useX` naming schema that is kind of "reserved" for React hooks these days. (We could still have this abstraction, but maybe it could be renamed and located in `./src/utils` ?)

The other option would be to pull your `useQuery` hook usage into this context and make this a full-on custom hook that abstracts those details away and truly lets you simply "use Quiz Categories". 👍 I think this is the route we should take.